From b0f30ab7ecdac98afa72a31371a393a73f855b3c Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Thu, 7 Apr 2022 21:39:26 +0800
Subject: [PATCH] Use system installed dependencies

Use the system versions of libva, tbb, OpenCV, ade and zlib.

Upstream-Status: Inappropriate
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>

---
 cmake/dependencies.cmake                | 75 -------------------------
 cmake/extra_modules.cmake               |  4 +-
 cmake/templates/OpenVINOConfig.cmake.in |  9 ---
 src/cmake/ie_parallel.cmake             |  2 +-
 src/plugins/intel_gpu/include/va/va.h   |  6 --
 tests/fuzz/src/CMakeLists.txt           |  3 +-
 thirdparty/CMakeLists.txt               | 26 ++++-----
 thirdparty/cnpy/CMakeLists.txt          |  2 +-
 8 files changed, 15 insertions(+), 112 deletions(-)
 delete mode 100644 src/plugins/intel_gpu/include/va/va.h

diff --git a/cmake/dependencies.cmake b/cmake/dependencies.cmake
index e2f941fcd..1fcc358ef 100644
--- a/cmake/dependencies.cmake
+++ b/cmake/dependencies.cmake
@@ -29,26 +29,6 @@ endif()
 
 if(CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME MATCHES Linux AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
     set(protoc_version "3.18.2")
-
-    RESOLVE_DEPENDENCY(SYSTEM_PROTOC_ROOT
-        ARCHIVE_LIN "protoc-${protoc_version}-linux-x86_64.tar.gz"
-        TARGET_PATH "${TEMP}/protoc-${protoc_version}-linux-x86_64"
-        SHA256 "42fde2b6044c1f74c7e86d4e03b43aac87128ddf57ac6ed8c4eab7a1e21bbf21"
-    )
-    debug_message(STATUS "host protoc-${protoc_version} root path = " ${SYSTEM_PROTOC_ROOT})
-
-    reset_deps_cache(SYSTEM_PROTOC)
-
-    find_host_program(
-        SYSTEM_PROTOC
-        NAMES protoc
-        PATHS "${SYSTEM_PROTOC_ROOT}/bin"
-        NO_DEFAULT_PATH)
-    if(NOT SYSTEM_PROTOC)
-        message(FATAL_ERROR "[ONNX IMPORTER] Missing host protoc binary")
-    endif()
-
-    update_deps_cache(SYSTEM_PROTOC "${SYSTEM_PROTOC}" "Path to host protoc for ONNX Importer")
 endif()
 
 if(ENABLE_INTEL_MYRIAD)
@@ -102,61 +82,6 @@ if(THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO")
         set(IE_PATH_TO_DEPS "${THIRDPARTY_SERVER_PATH}")
     endif()
 
-    if(WIN32 AND X86_64)
-        # TODO: add target_path to be platform specific as well, to avoid following if
-        RESOLVE_DEPENDENCY(TBB
-                ARCHIVE_WIN "tbb2020_20200415_win.zip"
-                TARGET_PATH "${TEMP}/tbb"
-                ENVIRONMENT "TBBROOT"
-                SHA256 "f1c9b9e2861efdaa01552bd25312ccbc5feeb45551e5f91ae61e29221c5c1479")
-        if(ENABLE_TBBBIND_2_5)
-            RESOLVE_DEPENDENCY(TBBBIND_2_5
-                    ARCHIVE_WIN "tbbbind_2_5_static_win_v1.zip"
-                    TARGET_PATH "${TEMP}/tbbbind_2_5"
-                    ENVIRONMENT "TBBBIND_2_5_ROOT"
-                    SHA256 "a67afeea8cf194f97968c800dab5b5459972908295242e282045d6b8953573c1")
-        else()
-            message(WARNING "prebuilt TBBBIND_2_5 is not available.
-    Build oneTBB from sources and set TBBROOT environment var before OpenVINO cmake configure")
-        endif()
-    elseif(ANDROID)  # Should be before LINUX due LINUX is detected as well
-        RESOLVE_DEPENDENCY(TBB
-                ARCHIVE_ANDROID "tbb2020_20200404_android.tgz"
-                TARGET_PATH "${TEMP}/tbb"
-                ENVIRONMENT "TBBROOT"
-                SHA256 "f42d084224cc2d643314bd483ad180b081774608844000f132859fca3e9bf0ce")
-    elseif(LINUX AND X86_64)
-        RESOLVE_DEPENDENCY(TBB
-                ARCHIVE_LIN "tbb2020_20200415_lin_strip.tgz"
-                TARGET_PATH "${TEMP}/tbb"
-                ENVIRONMENT "TBBROOT"
-                SHA256 "95b2f3b0b70c7376a0c7de351a355c2c514b42c4966e77e3e34271a599501008")
-        if(ENABLE_TBBBIND_2_5)
-            RESOLVE_DEPENDENCY(TBBBIND_2_5
-                    ARCHIVE_LIN "tbbbind_2_5_static_lin_v2.tgz"
-                    TARGET_PATH "${TEMP}/tbbbind_2_5"
-                    ENVIRONMENT "TBBBIND_2_5_ROOT"
-                    SHA256 "865e7894c58402233caf0d1b288056e0e6ab2bf7c9d00c9dc60561c484bc90f4")
-        else()
-            message(WARNING "prebuilt TBBBIND_2_5 is not available.
-    Build oneTBB from sources and set TBBROOT environment var before OpenVINO cmake configure")
-        endif()
-    elseif(LINUX AND AARCH64)
-        RESOLVE_DEPENDENCY(TBB
-                ARCHIVE_LIN "keembay/tbb2020_38404_kmb_lic.tgz"
-                TARGET_PATH "${TEMP}/tbb_yocto"
-                ENVIRONMENT "TBBROOT"
-                SHA256 "321261ff2eda6d4568a473cb883262bce77a93dac599f7bd65d2918bdee4d75b")
-    elseif(APPLE AND X86_64)
-        RESOLVE_DEPENDENCY(TBB
-                ARCHIVE_MAC "tbb2020_20200404_mac.tgz"
-                TARGET_PATH "${TEMP}/tbb"
-                ENVIRONMENT "TBBROOT"
-                SHA256 "ad9cf52e657660058aa6c6844914bc0fc66241fec89a392d8b79a7ff69c3c7f6")
-    else()
-        message(FATAL_ERROR "TBB is not available on current platform")
-    endif()
-
     update_deps_cache(TBBROOT "${TBB}" "Path to TBB root folder")
     if(EXISTS "${TBBROOT}/lib/cmake/TBB/TBBConfig.cmake")
         # oneTBB case
diff --git a/cmake/extra_modules.cmake b/cmake/extra_modules.cmake
index e33e4378f..f0cacdf6f 100644
--- a/cmake/extra_modules.cmake
+++ b/cmake/extra_modules.cmake
@@ -18,7 +18,7 @@ function(ie_generate_dev_package_config)
     configure_package_config_file("${OpenVINO_SOURCE_DIR}/cmake/templates/InferenceEngineDeveloperPackageConfig.cmake.in"
                                   "${CMAKE_BINARY_DIR}/InferenceEngineDeveloperPackageConfig.cmake"
                                   INSTALL_DESTINATION share # not used
-                                  PATH_VARS "OpenVINO_SOURCE_DIR;gflags_BINARY_DIR"
+                                  PATH_VARS "OpenVINO_SOURCE_DIR"
                                   NO_CHECK_REQUIRED_COMPONENTS_MACRO)
 
     configure_file("${OpenVINO_SOURCE_DIR}/cmake/templates/InferenceEngineConfig-version.cmake.in"
@@ -45,7 +45,7 @@ function(ov_generate_dev_package_config)
     configure_package_config_file("${OpenVINO_SOURCE_DIR}/cmake/templates/OpenVINODeveloperPackageConfig.cmake.in"
                                   "${CMAKE_BINARY_DIR}/OpenVINODeveloperPackageConfig.cmake"
                                   INSTALL_DESTINATION share # not used
-                                  PATH_VARS "OpenVINO_SOURCE_DIR;gflags_BINARY_DIR"
+                                  PATH_VARS "OpenVINO_SOURCE_DIR"
                                   NO_CHECK_REQUIRED_COMPONENTS_MACRO)
 
     configure_file("${OpenVINO_SOURCE_DIR}/cmake/templates/OpenVINOConfig-version.cmake.in"
diff --git a/cmake/templates/OpenVINOConfig.cmake.in b/cmake/templates/OpenVINOConfig.cmake.in
index 00e892f2b..7c686fa07 100644
--- a/cmake/templates/OpenVINOConfig.cmake.in
+++ b/cmake/templates/OpenVINOConfig.cmake.in
@@ -146,15 +146,6 @@ set(_ov_package_prefix_dir "${PACKAGE_PREFIX_DIR}")
 
 set(THREADING "@THREADING@")
 if((THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO") AND NOT TBB_FOUND)
-    set_and_check(_tbb_dir "@PACKAGE_IE_TBB_DIR@")
-    _ov_find_dependency(TBB
-                        COMPONENTS tbb tbbmalloc
-                        CONFIG
-                        PATHS ${TBBROOT}/cmake
-                              ${_tbb_dir}
-                        NO_CMAKE_FIND_ROOT_PATH
-                        NO_DEFAULT_PATH)
-
     set(install_tbbbind "@install_tbbbind@")
     if(install_tbbbind)
         set_and_check(_tbb_bind_dir "@PACKAGE_IE_TBBBIND_DIR@")
diff --git a/src/cmake/ie_parallel.cmake b/src/cmake/ie_parallel.cmake
index a036b019c..031f1ea48 100644
--- a/src/cmake/ie_parallel.cmake
+++ b/src/cmake/ie_parallel.cmake
@@ -34,7 +34,7 @@ function(set_ie_threading_interface_for TARGET_NAME)
             endforeach()
         endif()
         set(TBB_FOUND ${TBB_FOUND} PARENT_SCOPE)
-        set(TBB_IMPORTED_TARGETS ${TBB_IMPORTED_TARGETS} PARENT_SCOPE)
+        set(TBB_IMPORTED_TARGETS "tbb;tbbmalloc"  PARENT_SCOPE)
         set(TBB_VERSION ${TBB_VERSION} PARENT_SCOPE)
         if (NOT TBB_FOUND)
             set(THREADING "SEQ" PARENT_SCOPE)
diff --git a/src/plugins/intel_gpu/include/va/va.h b/src/plugins/intel_gpu/include/va/va.h
deleted file mode 100644
index 5c8a5c6f6..000000000
--- a/src/plugins/intel_gpu/include/va/va.h
+++ /dev/null
@@ -1,6 +0,0 @@
-// Copyright (C) 2018-2022 Intel Corporation
-// SPDX-License-Identifier: Apache-2.0
-//
-
-typedef cl_uint VASurfaceID;
-typedef void* VADisplay;
diff --git a/tests/fuzz/src/CMakeLists.txt b/tests/fuzz/src/CMakeLists.txt
index 5e6b7b168..a81c51bb7 100644
--- a/tests/fuzz/src/CMakeLists.txt
+++ b/tests/fuzz/src/CMakeLists.txt
@@ -10,14 +10,13 @@ add_custom_target(fuzz)
 FILE(GLOB tests "*-fuzzer.cc")
 
 add_subdirectory(../../../thirdparty/cnpy ${CMAKE_CURRENT_BINARY_DIR}/cnpy)
-add_subdirectory(../../../thirdparty/zlib ${CMAKE_CURRENT_BINARY_DIR}/zlib)
 
 foreach(test_source ${tests})
     get_filename_component(test_name ${test_source} NAME_WE)
     add_fuzzer(${test_name} ${test_source})
 
     target_link_libraries(${test_name} PRIVATE
-        openvino::runtime cnpy zlib)
+        openvino::runtime cnpy z)
 
     add_dependencies(fuzz ${test_name})
 
diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index 2d61c5fd7..97b372998 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -12,10 +12,9 @@ endif()
 
 add_subdirectory(ittapi)
 add_subdirectory(itt_collector EXCLUDE_FROM_ALL)
-add_subdirectory(zlib EXCLUDE_FROM_ALL)
 add_subdirectory(cnpy EXCLUDE_FROM_ALL)
 if(ENABLE_INTEL_GPU)
-    add_subdirectory(ocl)
+	#add_subdirectory(ocl)
 endif()
 
 add_subdirectory(xbyak EXCLUDE_FROM_ALL)
@@ -52,21 +51,16 @@ endif()
 add_library(ocv_hal INTERFACE)
 target_include_directories(ocv_hal INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/ocv")
 
-add_subdirectory(ade EXCLUDE_FROM_ALL)
 add_subdirectory(fluid/modules/gapi EXCLUDE_FROM_ALL)
-set_target_properties(ade fluid PROPERTIES FOLDER thirdparty)
-openvino_developer_export_targets(COMPONENT openvino_common TARGETS ade fluid)
+set_target_properties(fluid PROPERTIES FOLDER thirdparty)
+openvino_developer_export_targets(COMPONENT openvino_common TARGETS fluid)
 
-ov_install_static_lib(ade openvino_common)
 ov_install_static_lib(fluid openvino_common)
 
 #
 # Gflags
 #
 
-add_subdirectory(gflags EXCLUDE_FROM_ALL)
-openvino_developer_export_targets(COMPONENT openvino_common TARGETS gflags)
-
 #
 # Google Tests framework
 #
@@ -89,13 +83,13 @@ if(ENABLE_OV_PADDLE_FRONTEND OR ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_TF_FRONTEND
             set(Protobuf_DEBUG ON)
         endif()
         find_package(Protobuf 3.18.2 REQUIRED)
-        set(Protobuf_LITE_LIBRARIES protobuf::libprotobuf-lite)
-        set(Protobuf_LIBRARIES protobuf::libprotobuf)
-        set(SYSTEM_PROTOC protobuf::protoc)
+	set(Protobuf_LITE_LIBRARIES protobuf::libprotobuf-lite)
+	set(Protobuf_LIBRARIES protobuf::libprotobuf)
+	#set(SYSTEM_PROTOC protobuf::protoc)
         set(PROTOC_EXECUTABLE ${SYSTEM_PROTOC})
 
         foreach(target ${SYSTEM_PROTOC} ${Protobuf_LIBRARIES} ${Protobuf_LITE_LIBRARIES})
-            set_property(TARGET ${target} PROPERTY IMPORTED_GLOBAL TRUE)
+		#set_property(TARGET ${target} PROPERTY IMPORTED_GLOBAL TRUE)
         endforeach()
     else()
         add_subdirectory(protobuf EXCLUDE_FROM_ALL)
@@ -104,8 +98,8 @@ if(ENABLE_OV_PADDLE_FRONTEND OR ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_TF_FRONTEND
     # forward variables used in the other places
     set(SYSTEM_PROTOC ${SYSTEM_PROTOC} PARENT_SCOPE)
     set(PROTOC_EXECUTABLE ${PROTOC_EXECUTABLE} PARENT_SCOPE)
-    set(Protobuf_LIBRARIES ${Protobuf_LIBRARIES} PARENT_SCOPE)
-    set(Protobuf_LITE_LIBRARIES ${Protobuf_LITE_LIBRARIES} PARENT_SCOPE)
+    set(Protobuf_LIBRARIES ${Protobuf_LIBRARY} PARENT_SCOPE)
+    set(Protobuf_LITE_LIBRARIES ${Protobuf_LITE_LIBRARY} PARENT_SCOPE)
     set(Protobuf_INCLUDE_DIRS ${Protobuf_INCLUDE_DIRS} PARENT_SCOPE)
     set(Protobuf_IN_FRONTEND ON PARENT_SCOPE)
 
@@ -116,7 +110,7 @@ if(ENABLE_OV_PADDLE_FRONTEND OR ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_TF_FRONTEND
             set(link_type INTERFACE)
         endif()
         if(CMAKE_COMPILER_IS_GNUCXX OR OV_COMPILER_IS_CLANG)
-            target_compile_options(${target} ${link_type} -Wno-undef)
+		#target_compile_options(${target} ${link_type} -Wno-undef)
         endif()
     endforeach()
 endif()
diff --git a/thirdparty/cnpy/CMakeLists.txt b/thirdparty/cnpy/CMakeLists.txt
index 041031e92..793de9845 100644
--- a/thirdparty/cnpy/CMakeLists.txt
+++ b/thirdparty/cnpy/CMakeLists.txt
@@ -13,7 +13,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
                                           PRIVATE -Wno-all)
 endif()
 
-target_link_libraries(${TARGET_NAME} PUBLIC zlib)
+target_link_libraries(${TARGET_NAME} PUBLIC z)
 target_include_directories(${TARGET_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 
 set_target_properties(${TARGET_NAME} PROPERTIES FOLDER thirdparty)
-- 
2.35.1

